Source/WebCore/page/EventHandler.cpp

```diff
@@ -4283,6 +4281,93 @@ void EventHandler::defaultTextInputEventHandler(TextEvent& event)
         event.setDefaultHandled();
 }
 
+static const std::optional<KeyboardScrollingKey> keyboardScrollingKeyForKeyboardEvent(const KeyboardEvent& event)
```

Comment:
I think _ideally_ these would stay in KeyboardScrollingAnimator (if you need to use them from here, you can expose them as static methods on KeyboardScrollingAnimator, or as free functions). But maybe there is some reason they moved that I haven't realized yet?

Reply:
(and that will make this diff much smaller)

Reply:
Fixed